ci-operator/step-registry/telcov10n/metal-single-node-spoke/gitops/verify/telcov10n-metal-single-node-spoke-gitops-verify-commands.sh

```diff
@@ -37,11 +37,35 @@ function wait_for_managedcluster {
   wait_until_command_is_ok "oc get ns | grep -w '${SPOKE_CLUSTER_NAME}'" 10s 100
 }
 
+function try_to_recover_argocd_clusters_app {
+
+  echo "Cleaning and restoring ArgoCD apps..."
+
+  argo_clusters_app=$(mktemp --suffix=.json)
+  set -x
+  oc -n openshift-gitops get apps clusters -ojson | jq -r '
+  {
+    "apiVersion": .apiVersion,
+    "kind": .kind,
+    "metadata": {
+        "name": .metadata.name,
+        "namespace": .metadata.namespace
+    },
+    "spec": .spec
+  }' > ${argo_clusters_app}
+  oc replace -f ${argo_clusters_app}
```

Comment:
Should we prefer using `oc apply` instead of `replace`?

Reply:
Precisely what I do is to delete the part of the status that has been generated up to this point and pretend to install it again, so that it reads the Git repo again and rebuilds the whole status. I only do this, if the normal process fails, as an attempt to prevent the job from failing.